src/main/java/com/smartcar/sdk/SmartcarException.java

```diff
@@ -86,12 +86,13 @@ public String getRequestId() {
 
   public static SmartcarException Factory(final Response response) throws IOException {
     JsonObject body;
+    String bodyString = response.body().string();
     try {
-      body = gson.fromJson(response.body().string(), JsonObject.class);
```

Comment:
The issue here was because `response.body().string()` was called twice, once in try and once in catch. 

Source : https://github.com/square/okhttp/issues/1240